conans/assets/templates/new_v2_autotools.py

```diff
@@ -0,0 +1,241 @@
+import textwrap
+
+from conans.assets.templates.new_v2_cmake import source_cpp, source_h, test_main
+
+conanfile_lib = textwrap.dedent("""
+    import os
+
+    from conan import ConanFile
+    from conan.tools.gnu import AutotoolsToolchain, Autotools
+    from conan.tools.layout import basic_layout
+    from conan.tools.files import chdir
+
+
+    class {package_name}Conan(ConanFile):
+        name = "{name}"
+        version = "{version}"
+
+        # Optional metadata
+        license = "<Put the package license here>"
+        author = "<Put your name here> <And your email here>"
+        url = "<Package recipe repository url here, for issues about the package>"
+        description = "<Description of {package_name} here>"
+        topics = ("<Put some tag here>", "<here>", "<and here>")
+
+        # Binary configuration
+        settings = "os", "compiler", "build_type", "arch"
+        options = {{"shared": [True, False], "fPIC": [True, False]}}
+        default_options = {{"shared": False, "fPIC": True}}
+
+        exports_sources = "configure.ac", "Makefile.am", "src/*"
+
+        def config_options(self):
+            if self.settings.os == "Windows":
+                del self.options.fPIC
+
+        def layout(self):
+            basic_layout(self)
+
+        def generate(self):
+            at_toolchain = AutotoolsToolchain(self)
+            if self.options.shared:
+                at_toolchain.configure_args.extend(["--enable-shared", "--disable-static"])
+            else:
+                at_toolchain.configure_args.extend(["--disable-shared", "--enable-static",
+                                                    "--with-pic" if self.options.get_safe("fPIC", True) else "--without-pic"])
```

Comment:
This is a common pattern in Conan Center recipes and probably we should add built-in support for this in the toolchain, leaving this external here for the moment

Reply:
I have moved this to the toolchain